sdk/cosmos/azure-cosmos/azure/cosmos/_change_feed/feed_range_composite_continuation_token.py

```diff
@@ -0,0 +1,173 @@
+        overlapping_ranges = routing_provider.get_overlapping_ranges(collection_link, [self._current_token.feed_range])
+
+        if len(overlapping_ranges) == 1:
+            # merge,reusing the existing the feedRange and continuationToken
+            pass
+        else:
+            # split, remove the parent range and then add new child ranges.
+            # For each new child range, using the continuation token from the parent
+            self._continuation.popleft()
+            for child_range in overlapping_ranges:
+                self._continuation.append(
+                    CompositeContinuationToken(
+                        Range.PartitionKeyRangeToRange(child_range),
+                        self._current_token.token))
+
+            self._current_token = self._continuation[0]
+
+    async def handle_feed_range_gone_async(
+            self,
+            routing_provider: AsyncSmartRoutingMapProvider,
+            collection_link: str) -> None:
+        overlapping_ranges = \
+            await routing_provider.get_overlapping_ranges(
+                collection_link,
+                [self._current_token.feed_range])
+
+        if len(overlapping_ranges) == 1:
+            # merge,reusing the existing the feedRange and continuationToken
+            pass
+        else:
+            # split, remove the parent range and then add new child ranges.
+            # For each new child range, using the continuation token from the parent
+            self._continuation.popleft()
+            for child_range in overlapping_ranges:
+                self._continuation.append(
+                    CompositeContinuationToken(
+                        Range.PartitionKeyRangeToRange(child_range),
+                        self._current_token.token))
+
+            self._current_token = self._continuation[0]
+
+    def should_retry_on_not_modified_response(self) -> bool:
+        # when getting 304(Not Modified) response from one sub feed range,
+        # we will try to fetch for the next sub feed range
+        # we will repeat the above logic until we have looped through all sub feed ranges
+
+        # TODO: validate the response headers, can we get the status code
+        if len(self._continuation) > 1:
+            return self._current_token.feed_range != self._initial_no_result_range
+
+        return False
+
+    def _move_to_next_token(self) -> None:
+        first_composition_token = self._continuation.popleft()
+        # add the composition token to the end of the list
+        self._continuation.append(first_composition_token)
+        self._current_token = self._continuation[0]
+
+    def apply_server_response_continuation(self, etag) -> None:
+        self._current_token.update_token(etag)
+
+    def apply_not_modified_response(self) -> None:
+        if self._initial_no_result_range is None:
+            self._initial_no_result_range = self._current_token.feed_range
+
+    @property
+    def feed_range(self) -> FeedRange:
```

Comment:
after the self._feed_range being initialized, then it should only be allow to get but not set. If making the self._feed_range public, then it will also allow setting anywhere?

Reply:
discussed offline, will keep using the property